config/rbac/role.yaml

```diff
@@ -13,6 +13,18 @@ rules:
   - list
   - update
   - watch
+- apiGroups:
+  - ""
+  resources:
+  - pods
+  verbs:
+  - list
+- apiGroups:
+  - ""
+  resources:
+  - services
+  verbs:
```

Comment:
we might need endpoints & endpoint slices in addition to pods though if we go with service based discovery

Reply:
Not using enpoints and enpointslices currently, as you've stated in next comment here.